src/index.cpp

```diff
@@ -126,6 +130,89 @@ Index<T, TagT, LabelT>::Index(Metric m, const size_t dim, const size_t max_point
     }
 }
 
+template <typename T, typename TagT, typename LabelT>
+Index<T, TagT, LabelT>::Index(IndexConfig &index_config, std::unique_ptr<AbstractDataStore<T>> data_store)
```

Comment:
Is there a way to pass a flag (which can have a default value if needed to keep existing callers of such constructors as-is) to tell it whether we want to initialize `data_store` or not? Or some other approach?

Reply:
That should be possible, and we can achieve this by adding the flag to IndexConfig. But eventually we mean to retire the existing constructors(I think). Because 2 constructors, one for incremental and one for preloading existing index create confusion.